### PR TITLE
[refactor] #2011: More descriptive config params

### LINKED ISSUE #2011: Refactor configuration parameter names
Configuration parameters are completely incomprehensible. 

The word `MAILBOX` tells me nothing and we have it almost everywhere. 

Some misconfigurations need to be taken into account too.

### PR DESCRIPTION
**Pull request by ilchu**
Signed-off-by: Ilia Churin <churin.ilya@gmail.com>

### Description of the Change
Refactor names of some parameter config variables to make them more descriptive, such as the omnipresent `mailbox`.

### Issue
Resolves #2011.

### Benefits
Hopefully improved understanding of what the changed variables are meant to represent.

### Possible Drawbacks
Quite a bit of API changes with env variables being renamed all over.

### CODE REVIEW

File: p2p/src/network.rs [hunk 1]
```diff
@@ -69,8 +69,8 @@ where
     pub broker: Broker,
     /// Flag that stops listening stream
     finish_sender: Option<Sender<()>>,
-    /// Mailbox capacity
-    mailbox: u32,
+    /// Actor channel capacity
```
Comment (outoftardis) [inline p2p/src/network.rs:72]: how do you measure capacity? the number of blocks? I think this should be more descriptive than just repeating the `actor_channel_capacity` without underscores

File: cli/src/config.rs [hunk 2]
```diff
@@ -93,15 +93,15 @@ impl Default for Configuration {
 #[config(env_prefix = "IROHA_NETWORK_")]
 pub struct NetworkConfiguration {
     /// Actor mailbox size
```
Comment (outoftardis) [inline cli/src/config.rs:95]: mailbox is mentioned

File: core/src/block_sync.rs [hunk 3]
```diff
@@ -362,17 +362,17 @@ pub mod config {
         pub gossip_period_ms: u64,
         /// The number of blocks, which can be sent in one message.
         /// Underlying network (`iroha_network`) should support transferring messages this large.
-        pub batch_size: u32,
-        /// Mailbox size
-        pub mailbox: u32,
+        pub block_batch_size: u32,
+        /// Actor channel capacity
```
Comment (outoftardis) [inline core/src/block_sync.rs:366]: same comment here about a non-descriptive description

File: core/src/kura.rs [hunk 4]
```diff
@@ -555,9 +555,9 @@ pub mod config {
         /// Maximum number of blocks to write into single storage file
         #[serde(default = "default_blocks_per_storage_file")]
         pub blocks_per_storage_file: NonZeroU64,
-        /// Default mailbox size
-        #[serde(default = "default_mailbox_size")]
-        pub mailbox: u32,
+        /// Default actor channel capacity
```
Comment (outoftardis) [inline core/src/kura.rs:558]: same comment here about a non-descriptive description

File: core/src/sumeragi/config.rs [hunk 5]
```diff
@@ -35,13 +35,13 @@ pub struct SumeragiConfiguration {
...
-    pub commit_time_ms: u64,
+    pub commit_time_limit_ms: u64,
     /// Amount of time Peer waits for TxReceipt from the leader.
-    pub tx_receipt_time_ms: u64,
+    pub tx_receipt_time_limit_ms: u64,
     /// Limits to which transactions must adhere
     pub transaction_limits: TransactionLimits,
-    /// Mailbox size
-    pub mailbox: u32,
+    /// Actor channel capacity
```
Comment (outoftardis) [inline core/src/sumeragi/config.rs:43]: same comment here about a non-descriptive description

File: core/src/sumeragi/fault.rs [hunk 6]
```diff
@@ -93,8 +93,8 @@ where
     pub kura: AlwaysAddr<K>,
     /// [`iroha_p2p::Network`] actor address
     pub network: Addr<IrohaNetwork>,
-    /// Mailbox size
-    pub mailbox: u32,
+    /// Actor channel size
```
Comment (outoftardis) [inline core/src/sumeragi/fault.rs:96]: same comment here about a non-descriptive description

**Review by appetrosyan — DISMISSED** [review]

File: docs/source/references/config.md [hunk 7]
```diff
@@ -124,40 +124,40 @@ Has type `BlockSyncConfiguration`. Can be configured via environment variable `I
...
+  "BLOCK_BATCH_SIZE": 4,
+  "GOSSIP_PERIOD_MS": 10000
 }
 ```
 
-### `block_sync.batch_size`
+### `block_sync.actor_channel_capacity`
 
-The number of blocks, which can be sent in one message.
+Actor channel capacity
```
Comment (outoftardis) [inline docs/source/references/config.md:135]: need to say what is meant by capacity exactly

File: docs/source/references/config.md [hunk 8]
```diff
@@ -124,40 +124,40 @@ Has type `BlockSyncConfiguration`. Can be configured via environment variable `I
...
 ```json
-4
+100
 ```
 
-### `block_sync.gossip_period_ms`
+### `block_sync.block_batch_size`
 
-The time between sending request for latest block.
+The number of blocks, which can be sent in one message.
```
Comment (outoftardis) [inline docs/source/references/config.md:145]:
```suggestion
The number of blocks that can be sent in one message.
```

File: docs/source/references/config.md [hunk 9]
```diff
@@ -124,40 +124,40 @@ Has type `BlockSyncConfiguration`. Can be configured via environment variable `I
...
 ```json
-10000
+4
 ```
 
-### `block_sync.mailbox`
+### `block_sync.gossip_period_ms`
 
-Mailbox size
+The time between sending request for latest block.
```
Comment (outoftardis) [inline docs/source/references/config.md:155]: `between` implies there should be two events. do you mean the time between requests?

File: docs/source/references/config.md [hunk 10]
```diff
@@ -490,10 +490,20 @@ Has type `SumeragiConfiguration`. Can be configured via environment variable `IR
...
     }
   ],
-  "TX_RECEIPT_TIME_MS": 500
+  "TX_RECEIPT_TIME_LIMIT_MS": 500
 }
 ```
 
+### `sumeragi.actor_channel_capacity`
+
+Actor channel capacity
```
Comment (outoftardis) [inline docs/source/references/config.md:499]: non-descriptive

File: docs/source/references/config.md [hunk 11]
```diff
@@ -250,13 +250,23 @@ Has type `KuraConfiguration`. Can be configured via environment variable `IROHA_
...
   "BLOCK_STORE_PATH": "./blocks",
-  "INIT_MODE": "strict",
-  "MAILBOX": 100
+  "INIT_MODE": "strict"
 }
 ```
 
+### `kura.actor_channel_capacity`
+
+Default actor channel capacity
```
Comment (outoftardis) [inline docs/source/references/config.md:262]: non-descriptive

**Review by appetrosyan — APPROVED** [review]

**Review by outoftardis — APPROVED** [review]

**Pull request merged**